agent/src/main/java/com/code_intelligence/jazzer/instrumentor/HookMethodVisitor.kt

```diff
@@ -292,10 +264,18 @@ private class HookMethodVisitor(
         Opcodes.INVOKESPECIAL
     )
 
-    private fun findMatchingHook(hookType: HookType, owner: String, name: String, descriptor: String): Hook? {
-        val withoutDescriptorKey = "$hookType#$owner#$name"
-        val withDescriptorKey = "$withoutDescriptorKey#$descriptor"
-        return hooks[withDescriptorKey] ?: hooks[withoutDescriptorKey]
+    private fun findMatchingHooks(owner: String, name: String, descriptor: String): List<Hook> {
+        val result = HookType.values().map { hookType ->
```

Comment:
We should describe in the docs of `HookType` that the order of values is now important and defines the execution order.

Reply:
Let's do that based on discussing that in the team